dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/data/upload/UploadWorker.kt

```diff
@@ -96,8 +97,13 @@ internal class UploadWorker(
                 val batch = reader.read()
                 val batchMeta = reader.currentMetadata()
 
-                val success = consumeBatch(context, batch, batchMeta, uploader)
-                storage.confirmBatchRead(batchId) { confirmation ->
+                val uploadStatus = consumeBatch(context, batch, batchMeta, uploader)
+                val success = (uploadStatus is UploadStatus.Success)
+
+                storage.confirmBatchRead(
+                    batchId,
+                    RemovalReason.IntakeCode(uploadStatus.code)
+                ) { confirmation ->
```

Comment:
I probably missed that when we worked on v2, shouldn't we have the following? Otherwise seems like we only delete batches in cases of successful upload
``` 
confirmation.markAsRead(deleteBatch = !uploadStatus.shouldRetry)
```

Reply:
I actually wanted to ask you about this, good point @xgouchet